proxy/gateway/nodejs/lib/channel.js

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+'use strict';
+
+let EventEmitter = require('events').EventEmitter;
+let nano = require('nanomsg');
+let fs = require('fs');
+
+class Channel extends EventEmitter {
+  constructor() {
+    super();
+    this.socket = nano.socket('pair');
+    this.socket.on('data', data => this.emit('data', data));
+  }
+
+  connect(id) {
+    let addr = (typeof(id) === 'string') ? 'ipc://' + id : id.uri;
+    try {
+      fs.unlinkSync(addr.replace('ipc://', ''));
```

Comment:
Exactly...nanomsg uses Unix domain sockets on posix systems, so the URI is a path to the file. And for whatever reason they don't clean up the file (even though I found GitHub issues where they added code to do this, ultimately it doesn't appear to be happening) so we're doing it here.

Since Windows doesn't use files, I wrapped it in a `try` so that we'll try to delete a file that doesn't exist, we'll swallow the error, and we'll continue. Yes we can consider some other platform switching mechanism. I'll talk to you about this...